hack/ginkgo-e2e.sh

```diff
@@ -99,6 +99,11 @@ elif [[ "${KUBERNETES_PROVIDER}" == "aws" ]]; then
   auth_config=(
     "--auth_config=${HOME}/.kubernetes_auth"
   )
+elif [[ "${KUBERNETES_PROVIDER}" == "conformance_test" ]]; then
```

Comment:
`conformance_test` is not a provider; giving it a `cluster/util.sh` and treating it like a provider is only going to cause confusion.

Since you're really just using `KUBERNETES_PROVIDER` as a proxy for 'where's my auth config', how about this instead:
1. `ginkgo-e2e.sh` checks for an existing AUTH_CONFIG variable before doing the provider checks, and if it's set, use that to pass auth settings to the e2e binary.
2. `hack/conformance_test.sh` sets AUTH_CONFIG to point at stored auth info for the running cluster.

Side note: I'd actually like us to move away from using auth_config/cert_dir, and have providers use a utility to generate a standalone `.kubeconfig`, which they can drop in a location of their choosing. See #5154, which does adds said utility, and uses it for gce.